### ISSUE TITLE
GenericStructConstraintPropertiesTest fails in NativeAOT

### ISSUE THREAD

**Issue by jkotas** [dotnet/runtimelab#1563]
```
2021-09-17T16:21:24.9620888Z   [FAIL] System.Reflection.Tests.NullabilityInfoContextTests.GenericStructConstraintPropertiesTest(propertyName: "PropertyNullableEnabled", readState: NotNull, writeState: NotNull, type: typeof(int))
2021-09-17T16:21:24.9624192Z   System.ArgumentNullException : Value cannot be null. Parameter name: propertyInfo
2021-09-17T16:21:24.9625931Z      at System.Reflection.NullabilityInfoContext.Create(PropertyInfo) + 0x133
2021-09-17T16:21:24.9628191Z      at System.Reflection.Tests.NullabilityInfoContextTests.GenericStructConstraintPropertiesTest(String, NullabilityState, NullabilityState, Type) + 0x4b
2021-09-17T16:21:24.9631120Z      at System.Runtime.Tests!<BaseAddress>+0xf66455
2021-09-17T16:21:24.9759245Z      at System.InvokeUtils.CallDynamicInvokeMethod(Object, IntPtr, IntPtr, IntPtr, Object, Object[], BinderBundle, Boolean, Boolean) + 0x116
```
Disabled in #1562

**Comment by jkotas** [comment]
@MichalStrehovsky PTLA

**Comment by MichalStrehovsky** [comment]
Sigh, illinker bug, so we need to fix it there and then port the fix.

**Comment by MichalStrehovsky** [comment]
https://github.com/dotnet/linker/pull/2288 is the illinker fix.

**Comment by MichalStrehovsky** [comment]
Fixed #1599.
